compiler/integration-tests/circuits/double_verify/src/main.nr

```diff
@@ -0,0 +1,23 @@
+use dep::std;
```

Comment:
Do we need a separate circuit in here? Can't we just use `double_verify_proof` under the `execution_success` tests?

Reply:
True, however it looks like that test is severely out of date as it's using an incorrect proof length.

Reply:
I see 93 fields?

Reply:
Ah, I was looking at `double_verify_nested_proof`.

Reply:
Yeah so this accepts two proofs from `double_verify_proof_recursive` rather than `assert_statement_recurisve`. This would also be useful to test, only after a working double verify though. When we have a nested proof we pass the aggregation object as part of the proof. The separate circuit and larger proof size is necessary as whether we have an already recursive proof needs to be a circuit constant for the barretenberg verifier.

Reply:
I will make a PR to add some comments that elaborate on this

Reply:
Added some more info here (https://github.com/noir-lang/noir/pull/4567) as to avoid confusion in the future